langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingResponseBuilder.java

```diff
@@ -61,14 +65,22 @@ public void append(ChatCompletions completions) {
             return;
         }
 
-        FunctionCall functionCall = delta.getFunctionCall();
-        if (functionCall != null) {
-            if (functionCall.getName() != null) {
-                toolNameBuilder.append(functionCall.getName());
-            }
+        if (delta.getToolCalls() != null && !delta.getToolCalls().isEmpty()) {
+            ChatCompletionsToolCall toolCall = delta.getToolCalls().get(0);
```

Comment:
It takes only the first tool call, is this intentional? Parallel tool calls should already be supported, according to the [docs](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/function-calling?tabs=python#supported-models)

Reply:
It's coming from the OpenAI implementation, but indeed that's wrong.
I'm on it, it'll take some time as I'm adding some integration tests.

Reply:
I can now have multiple tools calls, but then the answers are streamed synchronously. I'm adding a full test suite, so it's easier to see.

Reply:
@jdubois awesome!

Could you please copy&adapt this tests into `AzureOpenAiChatModelIT` for sync model?

There are probably still 2 problems in `InternalAzureOpenAiHelper.functionCallFrom()`:
1. `chatRequestAssistantMessage.setFunctionCall(functionCallFrom(message));`: shouldn't `setToolCalls()` be used instead?
2. `return new ChatRequestFunctionMessage(nameFrom(message), toolExecutionResultMessage.text());`: shouldn't `ChatRequestToolMessage` be used instead?

Reply:
Indeed, I've found several issues there. I unfortunately don't have enough time right now, this is going to take me at least 2 weeks to fix.

Reply:
@jdubois no worries! It does not seem to be an urgent change, right?

Reply:
Should be good now!